src/commands/auth/device/login.ts

```diff
@@ -50,15 +50,15 @@ export default class Login extends SfdxCommand {
   };
 
   public async run(): Promise<AuthFields> {
-    if (await Prompts.shouldExitCommand(this.ux, this.flags.noprompt)) return {};
+    if (await Prompts.shouldExitCommand(this.ux, this.flags.noprompt as boolean)) return {};
```

Comment:
this command doesn't have a `--noprompt` flag defined, this could be replaced with:

```suggestion
    if (await Prompts.shouldExitCommand(this.ux, false)) return {};
```